app/helpers/application_helper.rb

```diff
@@ -13,4 +13,11 @@ def hash_array_json2csv(input_file, output_file, keys)
       end
     end
   end
+
+  def escape_json(hash)
+    # using json_escape to prevent XSS vulnerability
+    str = json_escape(hash.to_json) unless hash.class == 'String'
+    str = str.gsub!("\\", "\\\\\\") if str.include? "\\"
```

Comment:
Strings need to be json_escaped as well, I'd recommend that you use json_escape like the docs recommend rather than JSON.parse on the output.

https://apidock.com/rails/ERB/Util/json_escape